ddtrace/internal/writer.py

```diff
@@ -64,19 +67,39 @@ def flush_queue(self):
             log.error('error while filtering traces: {0}'.format(err))
             return
 
-        if not traces:
-            return
+        traces_flushed = len(traces)
+        traces_flushed_error = 0
 
         # If we have data, let's try to send it.
         traces_responses = self.api.send_traces(traces)
         for response in traces_responses:
             if isinstance(response, Exception) or response.status >= 400:
                 self._log_error_status(response)
+                traces_flushed_error += 1
             elif self._priority_sampler:
                 result_traces_json = response.get_json()
                 if result_traces_json and 'rate_by_service' in result_traces_json:
                     self._priority_sampler.set_sample_rate_by_service(result_traces_json['rate_by_service'])
 
+        # Dump statistics
```

Comment:
There are some default tags we should add.

1. If they configure `env` in the tracer, then we should attach the `env` tag here as well
  a. Note: we don't want to just take `tracer.tags`, we really want `env = tracer.tags.get('env'); if env: tags['env'] = env` (ish)
2. Since the metric names are pretty generic (good) we should tag them all with:
  a. `lang:python`
  b. `lang_version:3.7.x`
  c. `tracer_version:0.29.0`

Reply:
For 2., we should already have tags from the runtime metrics, e.g.  `lang_interpreter` and `lang_version`. I can add both `lang` and `tracer_version` as runtime tags too, that'd be ok I guess?

Reply:
If I'm not mistaken, `env` is already there since it's captured by runtime metrics tag collector.
https://github.com/DataDog/dd-trace-py/blob/master/ddtrace/internal/runtime/tag_collectors.py#L15

https://github.com/DataDog/dd-trace-py/pull/1069 might be all we need

Reply:
Yeah, seems like it: https://github.com/DataDog/dd-trace-py/blob/6f1b07b6e2d3c4c90e8ea57f41e9416eda5223b8/ddtrace/tracer.py#L332-L338